bdns/dns.go

```diff
@@ -230,54 +258,65 @@ func (dnsClient *DNSClientImpl) exchangeOne(ctx context.Context, hostname string
 		return nil, fmt.Errorf("Not configured with at least one DNS Server")
 	}
 
-	dnsClient.stats.Inc("Rate", 1)
-
 	// Randomly pick a server
 	chosenServer := dnsClient.servers[rand.Intn(len(dnsClient.servers))]
 
+	start := dnsClient.clk.Now()
 	client := dnsClient.dnsClient
-
+	qtypeStr := dns.TypeToString[qtype]
 	tries := 1
-	start := dnsClient.clk.Now()
-	msgStats.Inc("Calls", 1)
 	defer func() {
-		msgStats.TimingDuration("Latency", dnsClient.clk.Now().Sub(start))
+		result, authenticated := "failed", ""
+		if resp != nil {
+			result = dns.RcodeToString[resp.Rcode]
+			authenticated = fmt.Sprintf("%t", resp.AuthenticatedData)
+		}
+		dnsClient.totalLookupTime.With(prometheus.Labels{
+			"qtype":              qtypeStr,
+			"result":             result,
+			"authenticated_data": authenticated,
```

Comment:
This can only have two values (`true` or `false`) or are you talking about the label below for `retries`? It can have N values (up to whatever `maxRetries` is set to which is _probably_ okay. It might make more sense to have that as a separate histogram or counter or something but I wasn't super sure...